cin_validator/rules/cin2022_23/rule_8640.py

```diff
@@ -0,0 +1,73 @@
+from typing import Mapping
+
+import pandas as pd
+
+from cin_validator.rule_engine import (
+    CINTable,
+    IssueLocator,
+    RuleContext,
+    rule_definition,
+)
+from cin_validator.test_engine import run_rule
+
+# Get tables and columns of interest from the CINTable object defined in rule_engine/__api.py
+# Replace ChildIdentifiers with the table name, and LAChildID with the column name you want.
+
+CINdetails = CINTable.CINdetails
+ReasonForClosure = CINdetails.ReasonForClosure
+
+# define characteristics of rule
+@rule_definition(
+    code=8640,
+    module=CINTable.CINdetails,
+    message="CIN Reason for closure code invalid (see Reason for Closure table in CIN Census code set)",
+    affected_fields=[ReasonForClosure],
+)
+def validate(
+    data_container: Mapping[CINTable, pd.DataFrame], rule_context: RuleContext
+):
+    # Replace ChildIdentifiers with the name of the table you need.
+    df = data_container[CINdetails]
+
+    # implement rule logic as described by the Github issue. Put the description as a comment above the implementation as shown.
+
+    # Reason for Closure must be a valid reason for closure code value as shown in the list below
+
+    valid_reason = ["RC1","RC2","RC3","RC4","RC5","RC6","RC7","RC8","RC9"]
+
+    # Check if the Reason For Closure is not in the list of valid reasons and a value has been entered.
+
```

Comment:
Looks good, this got me wondering if Pandas had a built in method that is the inverse of isin, so you could avoid the ~, but I really can't think of one.